cassandra/cluster.py

```diff
@@ -3641,6 +3641,7 @@ def _set_new_connection(self, conn):
         with self._lock:
             old = self._connection
             self._connection = conn
+            self.refresh_schema()
 
         if old:
```

Comment:
Why is this necessary? The way I see it, the important bit of this PR is to drop relevant tablets after metadata update - but this changes when the metadata update is performed. To me it doesn't seem connected to this PR.

Reply:
When control connection is reconnected we need to reread metadata to trigger events for missing tables/hosts.

Reply:
Why do we need to do this immediately after reconnecting? Metadata refetch will be triggered eventually anyway.

Reply:
ASAIK there is no schedul to that, which means that it will not be triggered unless specific events come from server.
So, if no event come, it is going to stay as it is.

Reply:
`ControlConnection._try_connect` does refresh the schema as far as I can tell, and it is called by `_connect_host_in_lbp` which is called by `_reconnect_internal` which is called in `_reconnect` and in `_ControlReconnectionHandler.try_reconnect`

Reply:
indeed, removed.